test/oplog.js

```diff
@@ -358,6 +358,20 @@ test('oplog - multi append is atomic', async function (t) {
   await cleanup(storage)
 })
 
+test('oplog - header overflow', async function (t) {
+  const storage = testStorage()
+
+  const log = new Oplog(storage)
+  await log.open()
+
+  try {
+    await log.flush(Buffer.alloc(8192))
```

Comment:
i think 4kb is enough here

Reply:
(4kb + 1 to make it throw)

Reply:
The additional 8 bytes used by the oplog currently counts towards the overflow, so it'll throw at 4 KiB - 8 bytes. Is that correct?

Reply:
ya something like that. might be nice to test the actual limit just to see.